File: 128-longest-consecutive-sequence/longest-consecutive-sequence.ts
function longestConsecutive(nums: number[]): number {
    let numSet = new Set(nums);
    let longest = 0;

    while (numSet.size !== 0) {
        let num = numSet.values().next().value;
        numSet.delete(num);
        let [left, right] = [1, 1];

        while (numSet.has(num - left)) {
            numSet.delete(num - left);
            left += 1;
        }

        while (numSet.has(num + right)) {
            numSet.delete(num + right);
            right += 1;
        }
        longest = Math.max(left + right - 1, longest);
    }
    return longest;
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
"""
1. set()
2. 정렬 O(n log n)
"""

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        nums = sorted(list(set(nums)))
        max_cnt = 1
        cnt = 1

        for i in range(len(nums) - 1):
            if nums[i] + 1 == nums[i + 1]:
                cnt += 1
                max_cnt = max(max_cnt, cnt)
            else:
                cnt = 1

        return max_cnt